src/components/PriceLine.tsx

```diff
@@ -57,26 +58,51 @@ const PriceLine = ({
     if (!showBarrier) return null;
 
     return (
-        <div className='barrier-area' style={{ top: 0 }} ref={setDragLine} hidden={!visible}>
+        <div
+            className={classNames('barrier-area', { 'barrier-area--zero': isOverlappingWithPriceLine })}
+            style={{ top: 0 }}
+            ref={setDragLine}
+            hidden={!visible}
+        >
             <div
                 className={classNames('chart-line', 'horizontal', className || '', {
                     draggable,
                     dragging: isDragging,
                 })}
                 style={{
                     color: foregroundColor,
-                    backgroundImage:
-                        lineStyle && lineStyle !== 'solid' ? '' : `linear-gradient(to left, ${color} 90%, ${color}00`,
                 }}
             >
                 {showBarrierDragLine && (
-                    <div className='drag-line' style={{ borderTop: `${lineStyle} ${color} 1px` }} />
+                    <div
+                        className={classNames('drag-line', { 'drag-line--zero': isOverlappingWithPriceLine })}
+                        style={{
+                            borderTopStyle: lineStyle as React.CSSProperties['borderTopStyle'],
+                            width: `calc(100% - ${yAxiswidth}px + ${width}px)`,
+                        }}
+                    />
                 )}
                 <div className='draggable-area' />
-                <div className='drag-price' style={{ backgroundColor: color, width, opacity }}>
-                    <div className='price'>{priceDisplay}</div>
-                    {offScreen && offScreenDirection && (
-                        <PriceLineArrow offScreenDirection={offScreenDirection} color={color} />
+                <div className='draggable-area-wrapper'>
+                    <div
+                        className={classNames('drag-price')}
+                        style={{ width: isOverlappingWithPriceLine ? width : yAxiswidth, opacity, right: isOverlappingWithPriceLine ? yAxiswidth - width : 0 }}
+                    >
+                        <div className='drag-icon'>
+                            <div></div>
+                            <div></div>
+                            <div></div>
+                        </div>
+                        <div className={classNames('price', { 'price--zero': isOverlappingWithPriceLine })} style={{ right: isOverlappingWithPriceLine ? width + (priceDisplay.length * 8) - (!draggable ? 16 : 0) : 0}}>{priceDisplay}</div>
+                        <div />
```

Comment:
`!draggable ? 16 : 0 `===> `draggable ? 0 : 16`
But can be omitted)

Reply:
Or maybe it will make sense to lift up this calculation, like:
`const testVariable =  isOverlappingWithPriceLine ? width + (priceDisplay.length * 8) - (draggable ? 0 : 16) : 0}`
...
`style={{right: testVariable}}`

Reply:
I think it's fine, there's almost zero chance it will be reused anywhere else, so I didn't want to separate it. As for the first comment, I think having 0 as the first option is kind of counter-intuitive so I refer to put 0/null/undefined/etc options at the back)